src/main/java/com/example/auctionapp/dto/SearchRequest.java

```diff
@@ -0,0 +1,133 @@
+package com.example.auctionapp.dto;
+
+import com.example.auctionapp.customValidator.ColorConstraint;
+import com.example.auctionapp.customValidator.SizeConstraint;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+public class SearchRequest {
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    private Long subcategoryId;
+
+    @NotNull
+    @ColorConstraint
+    private String color;
+
+    @NotNull
+    @SizeConstraint
+    private String size;
+
+    @NotNull
+    @Min(-1)
```

Comment:
why min -1?

Reply:
It means default value. In case of -1, this filter won't be part of query